### PR TITLE
feat(cli): add new flag --max-file-size to control the max file size

### LINKED ISSUE #5989: Kubernetes scan skips Pods due to file size
### Discussed in https://github.com/Checkmarx/kics/discussions/5985

<div type='discussions-op-text'>

<sup>Originally posted by **MrSecure** November  4, 2022</sup>
I've been testing the Kubernetes scanning feature on a cluster which has ~325 pods, many which are in the Completed state, and awaiting cleanup.  However, because the `Pods.yaml` is larger than 5MB, *ALL* Pods get skipped.

Scan Command:

```bash
kics scan -q /usr/local/opt/kics/share/kics/assets/queries  \
  -p "kuberneter::*:*:*" -v --no-progress -o kicsout \
  --report-formats all --type Kubernetes

... omitted for clarity ...

Error: failed to read sources: failed to get file content: kicsout/kics-extract-kuberneter-11-03-2022/core/v1/Pods.yaml: file size limit exceeded
```

Partial output from `ls -l`:

```plain
6352203 Nov  3 14:22 kicsout/kics-extract-kuberneter-11-03-2022/core/v1/Pods.yaml
```

Pod Count

```bash
grep -c ^kind kicsout/kics-extract-kuberneter-11-03-2022/core/v1/Pods.yaml
323
```

5 MB Restriction Code:

https://github.com/Checkmarx/kics/blob/a52e458b41225af7ae343e293d1673daf4eee91b/pkg/kics/service.go#L132-L145

Has anyone else been hindered by this limit?   It looks like the intent is to prevent a resource exhaustion attack (see: https://github.com/Checkmarx/kics/issues/2156), however, there's no user control to adjust the limit.

Is there a way to specify a [Kuberneter path](https://github.com/Checkmarx/kics/blob/master/docs/integrations_kuberneter.md#kics-kuberneter-path-syntax) which excludes pods in specific states (or includes only pods in desired states)?   

What would be a better approach to handle large files - Process the first 5MB, rather than discarding the entire file because it's too large?   Present a configuration option to the user to set a higher limit?   Adjust Kuberneter library to split large files into smaller pieces? 

Can anyone lend a hand with creating a PR to adjust the limit in a sustainable manner?  (I'd prefer to avoid changing one arbitrary fixed limit for another)
</div>

### LINKED ISSUE #6306: Add an option to override the max size limit file
Currently, the max size limitation of a file is [5MB](https://github.com/Checkmarx/kics/blob/0bb5fa7b441ba1b8189757aba6506aeda8f01ee9/pkg/kics/service.go#L133)

When trying to scan a plan upon terraform post plan stage we failed due to this constraint.

**Suggestion**
Add the ability to override the max size limit of file with a specific flag (e.g. --max-file-size) to allow dynamically change this configuration.

We can work on it and contribute to the change, if we have the approval.

### PR DESCRIPTION
**Pull request by tomk-orca**
Closes #6306
Closes #5989

**Proposed Changes**
-implement a flag, control the size of the files scanned
-the flag should have as default value 5MB

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: internal/console/flags/scan_flags.go [hunk 1]
```diff
@@ -33,4 +33,6 @@ const (
 	DisableSecretsFlag     = "disable-secrets"
 	SecretsRegexesPathFlag = "secrets-regexes-path" //nolint:gosec
 	ExcludeGitIgnore       = "exclude-gitignore"
+	MaxFileSizeFlag        = "max-file-size" // orca-specifics
```
Comment (kaplanlior) [inline internal/console/flags/scan_flags.go:36]: When the PR is merged, the comment wouldn't be true any more (:

File: internal/console/assets/scan-flags.json [hunk 2]
```diff
@@ -195,5 +195,11 @@
     "shorthandFlag": "",
     "defaultValue": "false",
     "usage": "disables the exclusion of paths specified within .gitignore file"
+  },
+  "max-file-size": {
+    "flagType": "int",
+    "shorthandFlag": "",
+    "defaultValue": "5",
+    "usage": "max file size that allowed to scan in MB"
```
Comment (pereiramarco011) [inline internal/console/assets/scan-flags.json:203]: the description contains an error. I would advise changing it to "max scannable file size, in MB" or "max file size that will be scanned, in MB" or "max file size that is allowed to be scanned, in MB"

File: e2e/fixtures/assets/scan_help [hunk 3]
```diff
@@ -41,6 +41,7 @@ Flags:
                                       example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
       --input-data string             path to query input data files
   -b, --libraries-path string         path to directory with libraries (default "./assets/libraries")
+      --max-file-size int             max file size that allowed to scan in MB (default 5)
```
Comment (pereiramarco011) [inline e2e/fixtures/assets/scan_help:44]: change the description here too to pass the e2e tests

**Review by pereiramarco011 — CHANGES_REQUESTED** [review]
Left a couple of comments, other than that, and [Lior's comment](https://github.com/Checkmarx/kics/pull/6670/files#r1318135867) LGTM 😄 

Keep up the good work 👍

**Comment by JoaoCxMartins** [comment]
Hi @tomk-orca, any update on this pull request?

**Review by JoaoCxMartins — APPROVED** [review]

**Review by gabriel-cx — CHANGES_REQUESTED** [review]
@JoaoCxMartins kindly add this new command in this [file](https://github.com/Checkmarx/kics/blob/master/docs/commands.md), so the users can check it in our doc webpage.

**Review by gabriel-cx — APPROVED** [review]

File: internal/console/pre_scan.go [hunk 4]
```diff
@@ -157,6 +157,8 @@ func (console *console) preScan() {
 	cpu := consoleHelpers.GetNumCPU()
 	log.Info().Msgf("CPU: %.1f", cpu)
 
+	log.Info().Msgf("Max file size: %d Mb", flags.GetIntFlag(flags.MaxFileSizeFlag))
```
[the remark this answers is not in the document]
Comment (pereiramarco011) [reply internal/console/pre_scan.go:160]: I dont like the english on that and other references, it looks weird. I used chatgpt to give a few suggestions, here is the best:

Max file size permitted for scanning

what do you think? tis could be replaced on other references like the description of the flag and so on

**Review by JoaoCxMartins — DISMISSED** [review]

File: docs/commands.md [hunk 5]
```diff
@@ -83,6 +83,7 @@ Flags:
                                       example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
       --input-data string             path to query input data files
   -b, --libraries-path string         path to directory with libraries (default "./assets/libraries")
+      --max-file-size int             max file size that is allowed to be scanned, in MB (default 5)
```
Comment (gabriel-cx) [inline docs/commands.md:86]:
```suggestion
      --max-file-size int             max file size permitted for scanning, in MB (default 5)
```

File: e2e/fixtures/assets/scan_help [hunk 6]
```diff
@@ -43,6 +43,7 @@ Flags:
                                       example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
       --input-data string             path to query input data files
   -b, --libraries-path string         path to directory with libraries (default "./assets/libraries")
+      --max-file-size int             max file size that is allowed to be scanned, in MB (default 5)
```
Comment (gabriel-cx) [inline e2e/fixtures/assets/scan_help:46]:
```suggestion
      --max-file-size int             max file size permitted for scanning, in MB (default 5)
```

File: internal/console/assets/scan-flags.json [hunk 7]
```diff
@@ -207,5 +207,11 @@
     "shorthandFlag": "",
     "defaultValue": "false",
     "usage": "resolve the file reference, on OpenAPI files"
+  },
+  "max-file-size": {
+    "flagType": "int",
+    "shorthandFlag": "",
+    "defaultValue": "5",
+    "usage": "max file size that is allowed to be scanned, in MB"
```
Comment (gabriel-cx) [inline internal/console/assets/scan-flags.json:215]:
```suggestion
    "usage": "max file size permitted for scanning, in MB"
```

**Review by gabriel-cx — CHANGES_REQUESTED** [review]

**Review by gabriel-cx — DISMISSED** [review]

**Review by JoaoCxMartins — DISMISSED** [review]

**Review by pereiramarco011 — APPROVED** [review]

**Comment by asofsilva** [comment]
Hi @tomk-orca,

could you please add an E2E test where you can scan a file larger than 5MB to ensure that the file is not scanned? You can also ensure that, when using --max-file-size flag, the file is scanned. 

Thank you :)

**Comment by kaplanlior** [comment]
> Hi @tomk-orca,
> 
> could you please add an E2E test where you can scan a file larger than 5MB to ensure that the file is not scanned? You can also ensure that, when using --max-file-size flag, the file is scanned.
> 
> Thank you :)

We should add them for now, as he is probably unenviable for the E2E tests.

Please don't block due to this, we can even add a task to complete those as a separate issue.

**Review by cxMiguelSilva — APPROVED** [review]

**Review by cxMiguelSilva — APPROVED** [review]

**Pull request merged**